modules/nw-operator-vsphere-multiple-nic.adoc

```diff
@@ -0,0 +1,75 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_vsphere/ipi/installing-vsphere-installer-provisioned-network-customizations.adoc
+// * installing/installing_vsphere/upi/installing-vsphere-network-customizations.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-operator-vsphere-multiple-nic_{context}"]
+= Specifying multiple network interface cards (NICs)
+
+Before you install an {product-title} cluster on a {vmw-short} host, you can specify multiple network interface cards (NICs) for a networking implementation so that the {vmw-short} cloud controller manager (CCM) can select the appropriate NIC for a given networking situation. The {vmw-full} can use the NIC for managing pods and services on your cluster. For this configuration, you must specify internal and external Classless Inter-Domain Routing (CIDR) implementations in the {vmw-short} cloud controller manager (CCM) configuration. Each CIDR implementation lists an IP address range that the CCM uses to determine what NICs interact with traffic from internal and external networks.
+
+//Bare metal only?
+[IMPORTANT]
+====
+Each NIC must have access to link:https://quay.io/repository/[Quay.io] to obtain packages that are required for the installation of your cluster.
```

Comment:
This is not true. Often secondary networks will not have external access to things like Quay. IMHO the DNS configuration issue is separate from this and I think we should handle it using the bug Ross opened and not address it here. It's not clear to me that it's even valid to have a DNS server advertised in an OCP cluster that can't resolve OCP things, but that's a conversation for another time.

Reply:
We don't have a method of selecting which interface tries to pull baremetal-runtimecfg quay.io image?

Even if we don't have DNS issues, we probably need a way to select how the image is pulled.  Or maybe this would require disconnected installs.

Multiple networks opens us up to multiple DHCP misconfigs.   Unless we require DHCP disabled on all additional networks.  I think everyone uses static.

Reply:
Thank you both. I removed this note for now.

Reply:
Do I need to mention that if a customer sets multiple networks that they need to disable DHCP On unused networks to avoid conflicts on the preferred/active network?

Reply:
The default route defines how images are pulled. Unless you specifically added a route for quay, I guess, but I think that would be fragile since the quay IP address is not necessarily stable.

In any case, this is orthogonal to the problem we're trying to solve here, which is that CCM IP selection doesn't always work right in multi-nic environments. I would prefer to land this PR and discuss other possible multi-nic issues on the other bug.

Reply:
Which default route?  I don't think we've tested if podman supports Happy Eyeballs.  But we should be pulling IPv6 first, then IPv4.    We should be designing for DualStackIPv6Primary.